src/main/java/com/google/sps/utils/UserUtils.java

```diff
@@ -150,4 +151,18 @@ public static void addPoints(Entity userEntity, int numPoints, DatastoreService
     }
     datastore.put(userEntity);
   }
+
+  /**
+    Adds 20 points to a user for uploading if it has been more than a day since they last uploaded
+  */
+  public static void addUploadPoints(Entity userEntity, DatastoreService datastore){
+
+    QuizTimingPropertiesUtils utils = new QuizTimingPropertiesUtils();
+    if(userEntity.getProperty("lastUploadTime") == null || utils.isTimestampOutdated((long) userEntity.getProperty("lastUploadTime"))){
+        addPoints(userEntity, 20, datastore);
+        userEntity.setProperty("lastUploadTime", System.currentTimeMillis());
```

Comment:
I think the time gets updated when it's checked. I could be mistaken on that but if not then you wouldn't need line 163

Reply:
this is the only place this property is being set/updated; I'll rename this variable to be lastAwardedUploadPoints